storage/src/main/java/io/atomix/storage/buffer/ByteBufferBytes.java

```diff
@@ -120,6 +120,15 @@ public Bytes read(int position, byte[] bytes, int offset, int length) {
     return this;
   }
 
+  @Override
+  public Bytes read(final int offset, final ByteBuffer dst, final int dstOffset, final int length) {
+    for (int i = 0; i < length; i++) {
+      dst.put(dstOffset + i, (byte)readByte(offset + i));
+    }
+
+    return this;
+  }
```

Comment:
I'm wondering if it makes sense to replace this completely at some point with agrona which is more battle tested i think

Reply:
Agreed, and also replace Kryo (although if if with SBE or not we can discuss that)